CMakeLists.txt

```diff
@@ -54,6 +54,25 @@ target_include_directories(sick_safetyscanners2_node PUBLIC
   src/SickSafetyscannersLifeCycle.cpp
   src/utils/MessageCreator.cpp)
 
+# Register sick_safetyscanners2_component as a component
+add_library(sick_safetyscanners2_component SHARED
+  src/sick_safetyscanners2_node.cpp
```

Comment:
You shouldn't put the file with the main into the shared library. You would create the library without it; in add_executable you would only leave the file with the main, and you would then link the executable against the newy creaded shared library.